### DISCUSSION TITLE
Is there any sample code available how to use option patten with OpenTelemetry.Instrumentation.SqlClient using Dependency Injection.

### DISCUSSION THREAD

**Discussion by harshchaurasia2020** [open-telemetry/opentelemetry-dotnet#4406] [Q&A]
Is SqlClient supporting Option Pattern?
If yes, please provide some sample code to understand this feature in more easy way.
As I can see OpenTelemetry.Instrumentation.SqlClient has now dependency on OpenTelemetry.Api.ProviderBuilderExtensions.

References
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry.Api.ProviderBuilderExtensions
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/extending-the-sdk/README.md#registration-extension-method-guidance-for-library-authors

**Comment by utpilla** [comment]
You could refer the AspNetCore example app which makes use of Options +DI to configure `AspNetCoreInstrumentationOptions`:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/ef0c9217087bfca0d6fe9206ccde343eb4e5738c/examples/AspNetCore/appsettings.json#L30-L32

https://github.com/open-telemetry/opentelemetry-dotnet/blob/ef0c9217087bfca0d6fe9206ccde343eb4e5738c/examples/AspNetCore/Program.cs#L66

You could follow something similar for `SqlClientInstrumentationOptions`
